### PR TITLE
PATAND-243: Form step

### PR DESCRIPTION
**Pull request by MohamedSaleh-medable**
### Objective

* Task PATAND-243: Form step
* The Form step is just adding the other steps view in a linear layout vertical view .. the only thing added here is a divider between each step 

### How To Test

**Credentials:**

* Environment: Int-Dev
* Org: hybridstudy
* Study: Multi language study
* [mohamed_andela+enro1@medable.com](mailto:mohamed_andela+enro1@medable.com) / qpal1010

**Steps:**

1. Open the app and use the credentials above
1. Join Study
1. Open Form step

**Expected:**

1. The step UI is similar to the following Zeplin links

https://zpl.io/aMjxxRN

https://zpl.io/VxQqqwk

https://zpl.io/aBdAAW0

https://zpl.io/V49nnGM

https://zpl.io/aMjxxXr

https://zpl.io/VOpEEyl

https://zpl.io/VxQjrlk

https://zpl.io/am7xxee

### Branches

* PAT: task/PATAND-243
* Axon: task/PATAND-243
* RS: task/PATAND-243
* Cortex: development

### Links

* https://jira.devops.medable.com/browse/PATAND-243

Closes PATAND-243

### CODE REVIEW

File: backbone/src/main/java/org/researchstack/backbone/ui/step/body/FormBody.java [hunk 1]
```diff
@@ -50,18 +50,26 @@ public View getBodyView(int viewType, LayoutInflater inflater, ViewGroup parent)
 
         // Iterate through all steps and generate each compact view. Store each StepBody child in a
         // list to iterate over (e.g. within getStepResult())
-        for (QuestionStep questionStep : questionSteps) {
-            StepBody stepBody = createStepBody(questionStep);
-            if(!questionStep.isHidden()) {
+        for (int i = 0; i < questionSteps.size(); i++) {
+            StepBody stepBody = createStepBody(questionSteps.get(i));
+            if(!questionSteps.get(i).isHidden()) {
```
Comment (Gryzor) [inline backbone/src/main/java/org/researchstack/backbone/ui/step/body/FormBody.java:55]: Missing space `if (`

File: backbone/src/main/java/org/researchstack/backbone/ui/step/body/FormBody.java [hunk 2]
```diff
@@ -50,18 +50,26 @@ public View getBodyView(int viewType, LayoutInflater inflater, ViewGroup parent)
...
+                }
             }
             formStepChildren.add(stepBody);
         }
 
         return body;
     }
 
+    private View getDividerView(LayoutInflater inflater, ViewGroup parent) {
+        return inflater.inflate(R.layout.rsb_form_step_divider, parent, false);
```
Comment (Gryzor) [inline backbone/src/main/java/org/researchstack/backbone/ui/step/body/FormBody.java:70]: I'd inflate this divider once and use the same instance every time. You can either inflate this beforehand or simply do "lazy" like:

```
View getDividerView(...) {
if (this.divider == null) { 
   this.divider = inflater...
}

return this.divider
}
```

**Comment by alaa-alshaikh** [comment]
working for me :)

File: backbone/src/main/java/org/researchstack/backbone/ui/step/body/FormBody.java (hunk 2 above)
Comment (MohamedSaleh-medable) [reply backbone/src/main/java/org/researchstack/backbone/ui/step/body/FormBody.java:70]: Can't add the same view object to the linear layout body it cause 
java.lang.IllegalStateException: The specified child already has a parent. You must call removeView() on the child's parent first.

Comment (Gryzor) [reply backbone/src/main/java/org/researchstack/backbone/ui/step/body/FormBody.java:70]: Yeah, you're right. Nevermind, this shouldn't be an issue anyway, we can't have that many views in a form by design so...

**Pull request merged**